### PR TITLE
fix sometimes degu send coap message with garbage data

### PR DESCRIPTION
**Pull request by yoshidam-at**
fix remove garbage for not block message(single packet).

### CODE REVIEW

**Comment by yoshidam-at** [comment]
fix #22 2nd challenge.

**Review by naomitodori — APPROVED** [review]

**Review by atKeitaroTakahashi — APPROVED** [review]

File: moddegu.c [hunk 1]
```diff
@@ -39,6 +39,7 @@ STATIC mp_obj_t degu_get_shadow(void) {
 		printf("can't malloc\n");
 		return mp_const_none;
 	}
+	memset(payload, 0, MAX_COAP_MSG_LEN);
 
 	degu_coap_request("thing", COAP_METHOD_GET, payload, NULL);
```
Comment (ohsawa) [inline moddegu.c:44]: Good fixup. Though, it may worth design of degu_coap_request does not return payload size.

File: zcoap.c [hunk 2]
```diff
@@ -66,15 +68,21 @@ static int zcoap_request(int sock, u8_t *path, u8_t method, u8_t *payload, u16_t
...
 						BLOCK_WISE_TRANSFER_SIZE_GET);
 		}
-		else if (method == COAP_METHOD_POST || method == COAP_METHOD_PUT) {
-			coap_block_transfer_init(&blk_ctx, COAP_BLOCK_1024, *payload_len);
+		else if ((method == COAP_METHOD_POST || method == COAP_METHOD_PUT) &&
+			  *payload_len > COAP_BLOCK_THRESHOLD) {
+			r = coap_block_transfer_init(&blk_ctx, COAP_BLOCK_1024, *payload_len);
+			if (r != 0) {
+				LOG_ERR("failed to coap_block_transfer_init(%d)\n", r);
+				return 0;
```
Comment (ohsawa) [inline zcoap.c]: Its return value may handled by degu_coap_request() case statement, that handles "default". Does it prefer to set COAP_FAILED_TO_RECEIVE_RESPONSE(-1) ?

Comment (yoshidam-at) [reply zcoap.c]: I'll change return value to "code".
COAP_FAILED_TO_RECEIVE_RESPONSE(-1) means "cannot receive response", not system or function error.

**Review by ohsawa — APPROVED** [review]

**Pull request merged**